src/libraries/System.Net.ServerSentEvents/ref/System.Net.ServerSentEvents.cs

```diff
@@ -6,14 +6,20 @@
 
 namespace System.Net.ServerSentEvents
 {
+    public static partial class SseFormatter
+    {
+        public static System.Threading.Tasks.Task WriteAsync(System.Collections.Generic.IAsyncEnumerable<System.Net.ServerSentEvents.SseItem<string>> source, System.IO.Stream destination, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task WriteAsync<T>(System.Collections.Generic.IAsyncEnumerable<System.Net.ServerSentEvents.SseItem<T>> source, System.IO.Stream destination, System.Action<System.Buffers.IBufferWriter<byte>, System.Net.ServerSentEvents.SseItem<T>> itemFormatter, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
This ordering of IBufferWriter then SseItem looks wrong to me. We generally do source and then destination; the source here is the SseItem and the destination is the IBufferWriter.

Reply:
I was thinking of how serializers typically order their paramers, e.g. JsonConverter.Write accepts writer first and value second.

Reply:
cc: @bartonjs, @terrajobst

Reply:
I believe that we intentionally flipped the order in the callback to be (source, destiantion).

Notably, your PR here does not match what was approved.

Reply:
> Notably, your PR here does not match what was approved.

Yep, this is exploring possible amendments in the callback. Another one is passing the wrapping SseItem<T> instead of just T. Will notify fxdc once we're settled on the shape.

Reply:
> I believe that we intentionally flipped the order in the callback to be (source, destiantion).

While technically the value is the source and the writer is the destination, this ordering is not what we commonly use in the context of serialization. For example:

* [JsonConverter.Write(Utf8JsonWriter, T)](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverter-1.write?view=net-8.0)
* [JsonSerializer.SerializeAsync(Stream, T)](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serialize?view=net-8.0#system-text-json-jsonserializer-serialize(system-io-stream-system-object-system-type-system-text-json-serialization-jsonserializercontext))
* [DataContractSerializer.WriteObject(XmlWriter, object)](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datacontractserializer.writeobject?view=netframework-4.8.1#system-runtime-serialization-datacontractserializer-writeobject(system-xml-xmlwriter-system-object))

Given that the callback does represent a serialization callback, I would suggest we use the order as presented in this PR.

Reply:
https://youtu.be/ehUiLj2KvCg?t=5583

It started as writer first, then got flipped.  There was no objection.  And, wow, I hate listening to myself.

If you want to change it, you can bring it back for discussion.

Reply:
https://github.com/dotnet/runtime/pull/109832/commits/1b7a3e534a9d8a2f15ac155c2352c37740da3346

Reply:
I don't feel strongly about it; there are arguments to be made either way. The most important thing would be local consistency, meaning what do we expect a developer to do with the callback. If the 99% case is they're going to use JsonSerializer.Serialize in the callback, then having the order of arguments to the callback match that of the order of the method the callback will invoke seems reasonable. It'd be a stronger argument if you could just use JsonSerializer.Serialize as the callback (just passing the method group), but that won't work because it doesn't have an `IBufferWriter<byte>` overload and (with your changes) this is passing an `SseItem<T>` rather than a `T`.

Reply:
I think there's an argument to be made in principle, that the `IBufferWriter<byte>` parameter represents a stateful writer as opposed to a destination buffer. We typically place stateful writers as the first parameter and the value to be written as the second parameter. I won't die on this hill though.